lib/matobjplist.gi

```diff
@@ -595,138 +529,26 @@ InstallMethod( ZeroMatrix, "for two integers and a plist matrix",
-            for i in [1..nrrows] do
-                l[i] := Vector(rows{[(i-1)*rowlen+1..i*rowlen]},t);
-            od;
-        fi;
-    else
-        l := [];
-        nrrows := 0;
-    fi;
-    res := Objectify( TypeObj(m), [m![BDPOS],t,rowlen,l] );
+    res := Objectify( TypeObj(m), [m![BDPOS],rows,rows,l] );
     if not IsMutable(m) then
         SetFilterObj(res,IsMutable);
     fi;
     return res;
   end );
 
-
-############################################################################
-# A selection of list operations:
-############################################################################
-
-InstallOtherMethod( \[\], "for a plist matrix and a positive integer",
-#T Once the declaration of '\[\]' for 'IsMatrixObj' disappears,
-#T we can use 'InstallMethod'.
-  [ IsPlistMatrixRep, IsPosInt ],
-  function( m, p )
-    return m![ROWSPOS][p];
-  end );
-
-InstallMethod( \[\]\:\=,
-  "for a plist matrix, a positive integer, and a plist vector",
-  [ IsPlistMatrixRep and IsMutable, IsPosInt, IsPlistVectorRep ],
-  function( m, p, v )
-    m![ROWSPOS][p] := v;
-  end );
-
-InstallMethod( \{\}, "for a plist matrix and a list",
-  [ IsPlistMatrixRep, IsList ],
-  function( m, p )
-    local l;
-    l := m![ROWSPOS]{p};
-    return Objectify(TypeObj(m),[m![BDPOS],m![EMPOS],m![RLPOS],l]);
-  end );
-
-InstallMethod( Add, "for a plist matrix and a plist vector",
-  [ IsPlistMatrixRep and IsMutable, IsPlistVectorRep ],
-  function( m, v )
-    Add(m![ROWSPOS],v);
-  end );
-
-InstallMethod( Add, "for a plist matrix, a plist vector, and a pos. int",
-  [ IsPlistMatrixRep and IsMutable, IsPlistVectorRep, IsPosInt ],
-  function( m, v, p )
-    Add(m![ROWSPOS],v,p);
-  end );
-
-InstallMethod( Remove, "for a plist matrix",
-  [ IsPlistMatrixRep and IsMutable ],
-  m -> Remove( m![ROWSPOS] ) );
-
-InstallMethod( Remove, "for a plist matrix, and a position",
-  [ IsPlistMatrixRep and IsMutable, IsPosInt ],
-  function( m, p )
-    Remove( m![ROWSPOS],p );
-  end );
-#T must return the removed row if it was bound
-
-InstallMethod( IsBound\[\], "for a plist matrix, and a position",
-  [ IsPlistMatrixRep, IsPosInt ],
-  function( m, p )
-    return p <= Length(m![ROWSPOS]);
-  end );
-
-InstallMethod( Unbind\[\], "for a plist matrix, and a position",
-  [ IsPlistMatrixRep and IsMutable, IsPosInt ],
-  function( m, p )
-    if p <> Length(m![ROWSPOS]) then
-        ErrorNoReturn("Unbind\\[\\]: Matrices must stay dense, you cannot Unbind here");
-    fi;
-    Unbind( m![ROWSPOS][p] );
-  end );
-
-InstallMethod( \{\}\:\=, "for a plist matrix, a list, and a plist matrix",
-  [ IsPlistMatrixRep and IsMutable, IsList,
-    IsPlistMatrixRep ],
-  function( m, pp, n )
-    m![ROWSPOS]{pp} := n![ROWSPOS];
-  end );
-
-InstallMethod( Append, "for two plist matrices",
-  [ IsPlistMatrixRep and IsMutable, IsPlistMatrixRep ],
-  function( m, n )
-    Append(m![ROWSPOS],n![ROWSPOS]);
-  end );
-
+# FIXME: does ShallowCopy even make sense for a matrixobj???
```

Comment:
In `matobj2.gd`, we define `ShallowCopy` for objects in `IsRowListMatrix`.

Reply:
Spot on. Thing is, in my mind, `IsPlistMatrixRep` should not have been in `IsRowListMatrix`, but of course it still was when you wrote that review remark. I simply forgot to make that change... now done, and `IsRowPlistMatrixRep` added. And then I changed this `ShallowCopy` method to require `IsRowPlistMatrixRep`.